DOCS-6885 Add Quality Gates Setup, Search + Manage, and Explorer Documentation (#21089)
DataDog/documentation: content/en/quality_gates/explorer/facets.md

title: Quality Gates or Rule Execution Facets
kind: documentation
description: Learn about facets in the Quality Gates Explorer.
further_reading:
- link: 'quality_gates/explorer/'
  tag: 'Documentation'
  text: 'Learn about the Quality Gates Explorer'
- link: 'quality_gates/search/'
  tag: 'Documentation'
  text: 'Learn how to search your rules and executions'
---

{{< site-region region="gov" >}}
<div class="alert alert-warning">Quality Gates is not available in the selected site ({{< region-param key="dd_site_name" >}}) at this time.</div>
{{< /site-region >}}

{{< callout url="#" btn_hidden="true" >}}
Quality Gates is in public beta.
{{< /callout >}}

## Overview

Facets are user-defined tags and attributes from your rules or executions. They are useful for both [qualitative](#qualitative-facets) and [quantitative](#quantitative-measures) data analysis. Facets allow you to manipulate your rules or executions in search queries that appear on [dashboards][2] and in [notebooks][3].

[Creating facets](#creating-facets) is **not required** for [searching quality gates][5]. Autocomplete capabilities use existing facets, but also any input that matches incoming quality gates applies.

> The [Quality Gates Explorer][4] includes out-of-the-box facets such as `Test Status`, `Test Service`, `CI Status`, and `CI Provider`. You can use facets in the Quality Gates Explorer to [search for and filter quality gates][5].

Reviewer: These out of the box facets are leftover from Test, can we update with Quality Gate facets

Author: ```suggestion
The [Quality Gates Explorer][4] includes out-of-the-box facets such as `Status` and `Gate ID`. You can use facets in the Quality Gates Explorer to [search for and filter quality gates][5].
```

{{% /tab %}}
{{% tab "Rules" %}}

Navigate to [**CI** > **Quality Gate Executions**][101] and select **Rules** to access the list of [Quality Gates facets][102].

{{< img src="quality_gates/explorer/facets_rules.png" text="Facets list for rules on the Quality Gates Explorer" style="width:100%" >}}

[101]: https://app.datadoghq.com/ci/quality-gates/executions
[102]: /quality_gates/explorer/?tab=rules

{{% /tab %}}
{{< /tabs >}}

### Qualitative facets

Use qualitative facets when you need to:

- **Get relative insights** for values.
- **Count unique values**.
- Frequently **filter** your quality gates against particular values. For example, create a facet on an environment tag to scope troubleshooting down to development, staging, or production environments.<br>

**Note:** Although facets are not required for filtering on tags, defining facets for tags that you often use during investigations can help reduce your time to resolution.

### Quantitative measures

Use quantitative measures when you need to:

- **Aggregate** values from multiple quality gates.
- **Range filter** your quality gates.
- **Sort** your quality gates against that value.

#### Types

> Measures have either a long integer or double value for equivalent capabilities.

Reviewer: I believe this means that they can be used for the same purposes, but not sure. 

@liashenko do you know?

> Measures have either a long integer or double value for equivalent capabilities.

Reviewer 2: Yeah, basically you have the same features for integer and double measures.

> Measures have either a long integer or double value for equivalent capabilities.

Author: Does this rewording work?

```suggestion
Measures support long integers and double values.
```

#### Units

Measures support units (**time** in seconds or **size** in bytes) for handling of orders of magnitude at query time and display time. The unit is a property of the measure itself, not of the field.

> For example, consider a `duration` measure in nanoseconds. Suppose test runs from `service:A` have `duration:10000000`, meaning `10 milliseconds`. Supposed test runs from `service:B` have `duration:5000000`, meaning `5 milliseconds`. Use `duration:>2ms` to consistently query test run tags from both services at once. For more information about search queries, see [Search Syntax][6].

Reviewer: this example is from Testing - we can probably remove this "Units" section altogether, since we don't have the concept of adding measure tags to Quality Gate executions

> #### Units
> 
> Measures support units (**time** in seconds or **size** in bytes) for handling of orders of magnitude at query time and display time. The unit is a property of the measure itself, not of the field.
> 
> For example, consider a `duration` measure in nanoseconds. Suppose test runs from `service:A` have `duration:10000000`, meaning `10 milliseconds`. Supposed test runs from `service:B` have `duration:5000000`, meaning `5 milliseconds`. Use `duration:>2ms` to consistently query test run tags from both services at once. For more information about search queries, see [Search Syntax][6].

Author: ```suggestion
```

## Facet panel

> The search bar provides the most comprehensive set of interactions to filter and group your data. However, for many cases, the facet panel is a straightforward way to navigate into your data. Open a facet to see a summary of its content for the scope of the current query.

Reviewer: I think "open a facet", since the majority of them start collapsed. As long as the facet is open or expanded, you can see the summary of its content

The search bar and URL automatically reflect your selections from the facet panel.

- **Facets (qualitative)** show a list of unique values, and a count of quality gates that match each facet.
- **Measures (quantitative)** have a slider ranging between minimum and maximum values. Use the slider, or input numerical values, to scope the search query to different bounds.

### Grouping facets

Facets are grouped into meaningful themes in the facet list. Assigning or reassigning a group for a facet affects only the facet list, and has no impact on search or analytics.

### Filtering facets

Use the search facets field on the facet panel to filter the facet list and navigate to a particular facet. The search uses the facet display name and field name to scope results.

## Creating facets

> Creating a facet on a test run or pipeline execution attribute or tag is not a mandatory step to search for quality gates. Facets are useful if you wish to add a meaningful description to a specific test run or pipeline execution attribute, or if you want the attribute values to appear on the Facets list.

Reviewer: Remove mention of test run and pipeline execution. They can create a facet based on a rule execution

Author: ```suggestion
Creating a facet on a rule execution attribute is not a mandatory step to search for quality gates. Facets are useful if you wish to add a meaningful description to a specific rule execution attribute, or if you want the attribute values to appear on the Facets list.
```

### Creating facets from the facet list

You can create a facet directly from the facet panel by clicking **+ Add**.

{{< img src="quality_gates/explorer/facets.png" alt="Add a facet from the facet side panel" style="width:30%;">}}

Define the underlying field (key) name for this facet:

> - Use tag key name for infrastructure tags.

Reviewer: Is infrastructure tags relevant here? or can we remove it?

Author: ```suggestion
```
